java/ql/test/library-tests/frameworks/apache-collections/Test.java

```diff
@@ -0,0 +1,990 @@
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;orderedMap;;;MapKey of Argument[0];MapKey of ReturnValue;value"
+			OrderedMap out = null;
+			Map in = (Map)newTreeMapWithMapKey(source());
+			out = MapUtils.orderedMap(in);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;orderedMap;;;MapValue of Argument[0];MapValue of ReturnValue;value"
+			OrderedMap out = null;
+			Map in = (Map)newTreeMapWithMapValue(source());
+			out = MapUtils.orderedMap(in);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;populateMap;(Map,Iterable,Transformer);;Element of Argument[1];MapValue of Argument[0];value"
+			Map out = null;
+			Iterable in = (Iterable)newWithElement(source());
+			MapUtils.populateMap(out, in, (Transformer)null);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// Note it is tricky to get this to compile - the compiler thinks it is ambiguous
+			// which overload it should choose unless you put the generic types in correctly
+			// "org.apache.commons.collections4;MapUtils;true;populateMap;(MultiMap,Iterable,Transformer);;Element of Argument[1];MapValue of Argument[0];value"
+			MultiMap<Integer, String> out = null;
+			Iterable<String> in = (Iterable<String>)newWithElement(source());
+			MapUtils.populateMap(out, in, (Transformer<String, Integer>)null);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;predicatedMap;;;MapKey of Argument[0];MapKey of ReturnValue;value"
+			IterableMap out = null;
+			Map in = (Map)newTreeMapWithMapKey(source());
+			out = MapUtils.predicatedMap(in, null, null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;predicatedMap;;;MapValue of Argument[0];MapValue of ReturnValue;value"
+			IterableMap out = null;
+			Map in = (Map)newTreeMapWithMapValue(source());
+			out = MapUtils.predicatedMap(in, null, null);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;predicatedSortedMap;;;MapKey of Argument[0];MapKey of ReturnValue;value"
+			SortedMap out = null;
+			SortedMap in = (SortedMap)newTreeMapWithMapKey(source());
+			out = MapUtils.predicatedSortedMap(in, null, null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;predicatedSortedMap;;;MapValue of Argument[0];MapValue of ReturnValue;value"
+			SortedMap out = null;
+			SortedMap in = (SortedMap)newTreeMapWithMapValue(source());
+			out = MapUtils.predicatedSortedMap(in, null, null);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;putAll;;;ArrayElement of Argument[1];MapKey of Argument[0];value"
+			Map out = null;
+			Object[] in = (Object[])newWithArrayElement(source());
```

Comment:
Actually, we can probably remove most of these casts, right?
```suggestion
			Object[] in = newWithArrayElement(source());
```